core/src/blockchain/blockchain.rs

```diff
@@ -218,7 +218,12 @@ impl BlockChain {
         let grandparent_hash_of_new_block = parent_details_of_new_block.parent;
         let prev_best_hash = self.best_block_hash();
 
-        if parent_details_of_new_block.total_score + new_header.score() > self.best_proposal_block_detail().total_score
+        let hash = self.best_proposal_block_hash();
+        let header = new_block.header();
+        let parent_header = self.block_header(&parent_hash_of_new_block).expect("Invalid parent hash");
+        let best_header = self.block_header(&hash).expect("Invalid best hash");
+        if parent_header.number() + header.number() > self.best_proposal_block_detail().number
+            && parent_header.view() + header.view() < best_header.view()
```

Comment:
That's because the direct comparison of `number` and `view` sums are not equivalent to the previous score comparison. Unlike the score comparison, we should not compare the sum of `parent_header` and `header`.  The "if clause" checks if the new block can be a best block. For the condition, I thought the equivalent comparison is `(new_header.number(), self.best_proposal_block_detail().view()) > (self.best_proposal_block_detail().number(), new_header.view())`.

Reply:
Thanks for your advice, but could you check my implementation again? Still, 32 tests fail. There seems to be a problem in defining the function `.view()`, but I don't know what the problem is.